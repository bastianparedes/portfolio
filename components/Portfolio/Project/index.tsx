import React from 'react'
import { VscSignOut, VscGithub } from 'react-icons/vsc'
import Image from 'next/image'
import styles from './styles.module.scss'
import stylesRight from './stylesRight.module.scss'
import stylesLeft from './stylesLeft.module.scss'
import classNames from '../../../utils/classNames'

interface ProjectProps {
  index: number
  name: string
  link: string
  image: string
  github: string
  description: string
  technologies: string[]
}

const Project = ({
  index,
  name,
  link,
  image,
  github,
  description,
  technologies
}: ProjectProps): JSX.Element => {
  const stylesSide = index % 2 === 0 ? stylesRight : stylesLeft
  const imagePath = '/images/projects/' + image

  return (
    <div className={styles.projectCard}>
      {link !== '' ? (
        <a
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
          href={link}
          target="_blank"
          rel="noreferrer noopener"
        >
          <Image
            src={imagePath}
            alt={image}
            layout="fill"
            objectFit="contain"
          />
        </a>
      ) : (
        <div
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
        >
          <Image
            src={imagePath}
            alt={image}
            layout="fill"
            objectFit="contain"
          />
        </div>
      )}

      <h2 className={classNames(styles.projectName, stylesSide.projectName)}>
        {name}
      </h2>
      <div
        className={classNames(
          styles.projectDescriptionContainer,
          stylesSide.projectDescriptionContainer
        )}
      >
        <p className={styles.projectDescription}>{description}</p>
      </div>
      <span
        className={classNames(
          styles.projectTechnologies,
          stylesSide.projectTechnologies
        )}
      >
        {technologies.join(', ')}
      </span>
      <div
        className={classNames(styles.linksContainer, stylesSide.linksContainer)}
      >
        {link !== '' && (
          <a
            className={styles.linkWebsite}
            href={link}
            target="_blank"
            rel="noreferrer noopener"
          >
            <VscSignOut />
          </a>
        )}
        {github !== '' && (
          <a
            className={styles.linkGithub}
            href={github}
            target="_blank"
            rel="noreferrer noopener"
          >
            <VscGithub />
          </a>
        )}
      </div>
    </div>
  )
}

export default Project
