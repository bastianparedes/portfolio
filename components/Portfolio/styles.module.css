.projectsContainer {
  width: 100%;
  display: grid;
  justify-content: center;
  align-items: center;
  gap: 10rem;
  padding-bottom: 10rem;
  margin: auto;
}

@media screen and (max-width: 768px) {
  .projectsContainer {
    gap: 5rem;
    padding-bottom: 5rem;
  }
}

@media screen and (max-width: 480px) {
  .projectsContainer {
    gap: 2rem;
    padding-bottom: 2rem;
  }
}
