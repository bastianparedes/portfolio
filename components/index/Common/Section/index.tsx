import React from 'react';
import styles from './styles.module.scss';

interface SectionProps {
  id: string;
  title: string;
  children?: React.ReactNode;
}

const Section = (props: SectionProps): JSX.Element => {
  return (
    <section id={props.id} className={styles.section}>
      <h3 className={styles.title}>{props.title}</h3>
      {props.children}
    </section>
  );
};

export default Section;
