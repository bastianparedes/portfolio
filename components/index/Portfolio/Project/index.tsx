import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/legacy/image';
import { VscGithub, VscSignOut } from 'react-icons/vsc';

import styles from './styles.module.scss';
import stylesLeft from './stylesLeft.module.scss';
import stylesRight from './stylesRight.module.scss';

interface ProjectProps {
  index: number;
  name: string;
  link?: string;
  image?: string;
  github?: string;
  description: string;
  technologies: string[];
}

const Project = ({
  index,
  name,
  link,
  image,
  github,
  description,
  technologies
}: ProjectProps): JSX.Element => {
  const stylesSide = index % 2 === 0 ? stylesRight : stylesLeft;

  return (
    <div className={styles.projectCard}>
      {link !== undefined ? (
        <a
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
          href={link}
          rel="noreferrer noopener"
          target="_blank"
        >
          {image !== undefined && (
            <Image
              alt={image}
              layout="fill"
              objectFit="contain"
              src={'/images/projects/' + image}
            />
          )}
        </a>
      ) : (
        <div
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
        >
          {image !== undefined && (
            <Image
              alt={image}
              layout="fill"
              objectFit="contain"
              src={'/images/projects/' + image}
            />
          )}
        </div>
      )}

      <h2 className={classNames(styles.projectName, stylesSide.projectName)}>
        {name}
      </h2>
      <div
        className={classNames(
          styles.projectDescriptionContainer,
          stylesSide.projectDescriptionContainer
        )}
      >
        <p className={styles.projectDescription}>{description}</p>
      </div>
      <span
        className={classNames(
          styles.projectTechnologies,
          stylesSide.projectTechnologies
        )}
      >
        {technologies.join(', ')}
      </span>
      <div
        className={classNames(styles.linksContainer, stylesSide.linksContainer)}
      >
        {link !== undefined && (
          <a
            className={styles.linkWebsite}
            href={link}
            rel="noreferrer noopener"
            target="_blank"
          >
            <VscSignOut />
          </a>
        )}
        {github !== undefined && (
          <a
            className={styles.linkGithub}
            href={github}
            rel="noreferrer noopener"
            target="_blank"
          >
            <VscGithub />
          </a>
        )}
      </div>
    </div>
  );
};

export default Project;
