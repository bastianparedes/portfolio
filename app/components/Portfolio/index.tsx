import React from 'react';

import Project from './Project';
import styles from './styles.module.scss';
import constants from '../../../config/constants';
import { Section } from '../Common';

const Portfolio = (): JSX.Element => {
  return (
    <Section id="portfolio" title="Portfolio">
      <div className={styles.projectsContainer}>
        {constants.PORTFOLIO.PROJECTS.map((info, index) => (
          <Project
            description={info.DESCRIPTION}
            github={info.GITHUB}
            image={info.IMAGE}
            index={index}
            key={info.NAME}
            link={info.LINK}
            name={info.NAME}
            technologies={info.TECHNOLOGIES}
          />
        ))}
      </div>
    </Section>
  );
};

export default Portfolio;
