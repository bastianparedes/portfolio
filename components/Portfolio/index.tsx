import React from 'react'
import { Section } from '../Common'
import styles from './styles.module.css'
import Project from './Project'
import constants from '../../config/constants.json'

const Portfolio = (): JSX.Element => {
  return (
    <Section id="portfolio" title="Portfolio">
      <div className={styles.projectsContainer}>
        {constants.PORTFOLIO.PROJECTS.map((info, index) => (
          <Project
            key={info.NAME}
            index={index}
            name={info.NAME}
            link={info.LINK}
            image={info.IMAGE}
            github={info.GITHUB}
            description={info.DESCRIPTION}
            technologies={info.TECHNOLOGIES}
          />
        ))}
      </div>
    </Section>
  )
}

export default Portfolio
