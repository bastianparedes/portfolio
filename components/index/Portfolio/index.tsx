import React from 'react';

import constants from '../../../config/constants.json';
import { Section } from '../Common';
import Project from './Project';
import styles from './styles.module.scss';

const Portfolio = (): JSX.Element => {
  return (
    <Section id="portfolio" title="Portfolio">
      <div className={styles.projectsContainer}>
        {constants.PORTFOLIO.PROJECTS.map((info, index) => (
          <Project
            key={info.NAME}
            index={index}
            name={info.NAME}
            link={info.LINK}
            image={info.IMAGE}
            github={info.GITHUB}
            description={info.DESCRIPTION}
            technologies={info.TECHNOLOGIES}
          />
        ))}
      </div>
    </Section>
  );
};

export default Portfolio;
