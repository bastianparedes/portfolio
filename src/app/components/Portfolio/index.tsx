import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/image';
import { VscGithub, VscSignOut } from 'react-icons/vsc';

import projects from './projects';
import styles from './styles.module.scss';
import stylesLeft from './stylesLeft.module.scss';
import stylesRight from './stylesRight.module.scss';
import { Section } from '../Common';

const Portfolio = () => {
  return (
    <Section id="portfolio" title="Portfolio">
      <div className={styles.projectsContainer}>
        {projects.map((project, index) => {
          const stylesSide = index % 2 === 0 ? stylesRight : stylesLeft;
          return (
            <div className={styles.projectCard} key={project.name}>
              {project.link !== undefined ? (
                <a
                  className={classNames(
                    styles.imageContainer,
                    stylesSide.imageContainer
                  )}
                  href={project.link}
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  <Image
                    alt={project.name}
                    placeholder="blur"
                    src={project.image}
                    style={{
                      height: '100%',
                      objectFit: 'contain',
                      width: '100%'
                    }}
                  />
                </a>
              ) : (
                <div
                  className={classNames(
                    styles.imageContainer,
                    stylesSide.imageContainer
                  )}
                >
                  <Image
                    alt={project.name}
                    placeholder="blur"
                    src={project.image}
                    style={{
                      height: '100%',
                      objectFit: 'contain',
                      width: '100%'
                    }}
                  />
                </div>
              )}
              <h2
                className={classNames(
                  styles.projectName,
                  stylesSide.projectName
                )}
              >
                {project.name}
              </h2>
              <div
                className={classNames(
                  styles.projectDescriptionContainer,
                  stylesSide.projectDescriptionContainer
                )}
              >
                <p className={styles.projectDescription}>
                  {project.description}
                </p>
              </div>
              <span
                className={classNames(
                  styles.projectTechnologies,
                  stylesSide.projectTechnologies
                )}
              >
                {project.technologies.join(', ')}
              </span>
              <div
                className={classNames(
                  styles.linksContainer,
                  stylesSide.linksContainer
                )}
              >
                {project.link !== undefined && (
                  <a
                    className={styles.linkWebsite}
                    href={project.link}
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    <VscSignOut />
                  </a>
                )}
                {project.github !== undefined && (
                  <a
                    className={styles.linkGithub}
                    href={project.github ?? undefined}
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    <VscGithub />
                  </a>
                )}
              </div>
            </div>
          );
        })}
      </div>
    </Section>
  );
};

export default Portfolio;
