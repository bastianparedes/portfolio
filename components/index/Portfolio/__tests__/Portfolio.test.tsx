import React from 'react';
import Portfolio from '..';
import { render } from '@testing-library/react';
import { Section } from '../../Common';

jest.mock('../../Common');
jest.mock('../Project');

describe('<Portfolio />', () => {
  (Section as jest.Mock).mockImplementation(({ children }) => <>{children}</>);

  it('should render', () => {
    const { container } = render(<Portfolio />);
    expect(container).toMatchSnapshot();
  });
});
