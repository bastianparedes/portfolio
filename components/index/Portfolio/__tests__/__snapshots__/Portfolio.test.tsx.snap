// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolio /> should render 1`] = `
<div>
  <div
    class="projectsContainer"
  />
</div>
`;
